src/libs/UnreadIndicatorUpdater/index.js

```diff
@@ -1,38 +1,29 @@
-import {InteractionManager} from 'react-native';
 import Onyx from 'react-native-onyx';
 import _ from 'underscore';
 import * as ReportUtils from '@libs/ReportUtils';
-import CONST from '@src/CONST';
+import Navigation, {navigationRef} from '@navigation/Navigation';
 import ONYXKEYS from '@src/ONYXKEYS';
 import updateUnread from './updateUnread/index';
 
-let previousUnreadCount = 0;
+let allReports = [];
+
+const triggerUnreadUpdate = () => {
+    const currentReportID = navigationRef.isReady() ? Navigation.getTopmostReportId() : '';
```

Comment:
Have you tested this `''` case? I understand the defence-in-depth strategy, but I'm not convinced by this specific implementation of it.

Maybe...

```ts
const currentReportID = navigationRef.isReady() ? Navigation.getTopmostReportId() : null;

const unreadReports = _.filter(allReports, (report) => {
    const isAccessible = currentReportID !== null ? ReportUtils.shouldReportBeInOptionList(report, currentReportID) : true;
    
    return ReportUtils.isUnread(report) && isAccessible;
});

```

Reply:
or ` : false`, depending on what we want to do in the corner case

Reply:
This is in accordance with how we're saving `currentReportID` in `CurrentReportIdContext`

Reply:
Ok